### DISCUSSION TITLE
feature for document history of any pdf being accessible by creating new namespace for every new document

### DISCUSSION THREAD

**Discussion by Aditya1404Sal** [mayooear/ai-pdf-chatbot-langchain#53] [Ideas]
correct me if i'm wrong 
let me just start off by stating the change in depth 
so a little preface , switching between different pdf's of varying sizes was the first thing i did , but as it turns out i had to repeatedly run the ingest script in order to play with the chat capabillities of the different documents and  it resulted in the duplication of the vectors on the exact same namespace i was thinking wouldn't it be more efficient to assign a namespace dynamically if a new document is detected ? 
plus a change to the landing page to implement a history section just like the chat-gpt interface. 
if there's somebody working on something simillar please let me know and i'll try contributing to the same

**Comment by mayooear** [comment]
Yeh that's covered in a PR I think someone recently made [here](https://github.com/mayooear/gpt4-pdf-chatbot-langchain/pulls), so if you can contribute to that, would be much appreciated.
